tdm-reader/nitdmreader.yml

```diff
@@ -0,0 +1,512 @@
+        type: integer
+        description: Number of pixels on the y axis which is the number of points that can be displayed
+        example: 1000
+        default: 0
+        format: int32
+      xmin:
+        type: number
+        default: 0
+        format: double
+        description: Minimum X value of the display X axis.
+        example: -1
+      xmax:
+        type: number
+        default: 0
+        format: double
+        description: Maximum X value of the display X axis.
+        example: 1
+      ymin:
+        type: number
+        default: 0
+        format: double
+        description: Minimum Y value of the display Y axis.
+        example: -1
+      ymax:
+        type: number
+        default: 0
+        format: double
+        description: Maximum Y value
+        example: 1
+      showLines:
+        type: boolean
+        description: Decimate to line style graph.
+        default: true
+      showPoints:
+        type: boolean
+        default: false
+        description: Decimate to point style graph.
+  OneChannelSpecifier:
+    description: Specification for a channel of data
+    properties:
+      fileId:
+        description: file descriptor
+        type: string
+        example: "5b8453dba8808c22b0f99023"
+      groupName:
+        type: string
+        description: group name
+        example: Group 1
+      channelName:
+        type: string
+        description: channel name
+        example: Channel 1
+    required: 
+      - fileId
+      - groupName
+      - channelName
+  ChannelSpecifications:
+    description: Specification for an array of groups of channels potentially with x-y relationships.
+    properties:
+      xyChannels:
+        type: array
+        items:
+          description: Y channel(s) and an optional x channel
+          properties:
+            xChannel:
+              $ref: '#/definitions/OneChannelSpecifier'
+            yChannels:
+              description: 'Channels to be treated as y channel, associated with x channel'
+              type: array
+              items:
+                $ref: '#/definitions/OneChannelSpecifier'
+          required:
+            - y
+  ChannelSpecificationsAndWindow:
+    description: Channel specifications and data window.
+    properties:
+      channelSpecificationsAndWindow:
+        properties:
+          channelSpecifications:
+            $ref: '#/definitions/ChannelSpecifications'
+          dataWindow:
+            $ref: '#/definitions/DataWindow'
+        required:
+          - channelSpecifications
+  OneChannelData:
+    properties:
+      dataType:
+        type: string
+        description: Native data type of data
+      numericData:
+        type: array
+        description: channel data
+        items:
+          type: number
+      stringData:
+        type: array
+        description: channel data
+        items:
+          type: string
+      decimated:
+        type: boolean
+        description: is returned data decimated
```

Comment:
Specifies whether returned data is decimated [though again, there might be a clearer way to describe decimation]